src/frontend/src/catalog/table_catalog.rs

```diff
@@ -155,6 +154,9 @@ pub struct TableCatalog {
 
     /// description of table, set by `comment on`.
     pub description: Option<String>,
+
+    /// Output indices, only used by replicated state table.
+    pub output_indices: Vec<usize>,
```

Comment:
> If the output_indices is only used by the replicated state table, what about adding them later?
We still need it here. Backfill reads from storage table with output indices. Similarly here, we also need to read with output indices. We can either include that field in backfill or state table. But since replicated state table will use it, I think it's better to add it in state table and avoid a refactor later.

> BTW, table catalog with an ouput_indices seems a little weird. Can we maintain this field in the stream scan proto?
I prefer not to. Main reason is that replicated state table will be constructed from table catalog.
We could pass it in from stream scan, but IMO state table's meta data (e.g. output_indices) should be entirely contained in table catalog.